packages/menu/src/uxp-menu-item.css

```
@@ -11,11 +11,36 @@ governing permissions and limitations under the License.
 */
 
 /* write uxp style overrides */
+:host {
+    display: flex;
+    flex-direction: row; /* Main layout direction */
+}
 
-/* Workaround for - https://jira.corp.adobe.com/browse/UXP-21329 */
-#selected {
-    padding-left: -8px;
+:host {
+    min-height:  var(--mod-menu-item-min-height, var(--spectrum-menu-item-min-height) );
+    padding-bottom: var(--mod-menu-item-bottom-edge-to-text, var(--spectrum-menu-item-bottom-edge-to-text) );
+    padding-right: var(--mod-menu-item-label-inline-edge-to-content, var(--spectrum-menu-item-label-inline-edge-to-content) );
+    padding-top: var(--mod-menu-item-top-edge-to-text, var(--spectrum-menu-item-top-edge-to-text) );
```

```suggestion
        padding-left: var(
        --mod-menu-item-label-inline-edge-to-content,
        var(--spectrum-menu-item-label-inline-edge-to-content)
    );
```
Padding left also has to be there to achieve correct.